### ISSUE TITLE
Need more configurable chain parameters

### ISSUE THREAD

**Issue by elric1** [tezedge/tezedge#655]
When starting tezedge, I see that you can specify `--network=foo` which selects a series of parameters such as the chain name, protocol, genesis block, etc. An example from the code of such a setting can be found in the code here:
```
    env.insert(TezosEnvironment::Sandbox, TezosEnvironmentConfiguration {
        genesis: GenesisChain {
            time: "2018-06-30T16:07:32Z".to_string(),
            block: "BLockGenesisGenesisGenesisGenesisGenesisf79b5d1CoW2".to_stri
ng(),
            protocol: "PtYuensgYBb3G3x1hLLbCmcav8ue8Kyd2khADcL5LsT5R1hcXex".to_s
tring(),
        },
        bootstrap_lookup_addresses: vec![],
        version: "SANDBOXED_TEZOS".to_string(),
        protocol_overrides: ProtocolOverrides {
            user_activated_upgrades: vec![],
            user_activated_protocol_overrides: vec![],
        },
        enable_testchain: false,
        patch_context_genesis_parameters: Some(PatchContext {
            key: "sandbox_parameter".to_string(),
            json: r#"{ "genesis_pubkey": "edpkuSLWfVU1Vq7Jg9FucPyKmma6otcMHac9zG
4oU1KMHSTBpJuGQ2" }"#.to_string(),
        }),
    });
```
For the tezos-k8s project, we need to be able to set some of these parameters arbitrarily. Here's the configuration section that we are using:
```
node_config_network:
  activation_account_name: baker0
  chain_name: SANDBOXED_TEZOS
  genesis:
    block: BLockGenesisGenesisGenesisGenesisGenesisf79b5d1CoW2
    protocol: PtYuensgYBb3G3x1hLLbCmcav8ue8Kyd2khADcL5LsT5R1hcXex
    timestamp: '2018-06-30T16:07:32Z'
```
We use the "activation account" to find the `genesis_pubkey` that ends up in the config file. But, as you can see, we need for various other fields from the above code block to be configurable.

**Comment by tizoc** [comment]
Hi @elric1, what are you using for Octez? a custom network in the context file? can you show me the custom network value you are using?

**Comment by tizoc** [comment]
Hello @elric1. This is now implemented and released in `v1.6.5`.

You should now be able to use custom networks by specifying the `--network=custom --custom-network-file=path/to/network-config.json` command-line flags. The `network-config.json` files must contain a json dictionary with the custom network configuration as the `"network"` value. The same format used in the Octez config file is expected.

Let me know if it solves your issue.

**Comment by tizoc** [comment]
Closing this one, I think we can consider it solved for now.
